src/transformers/modeling_utils.py

```diff
@@ -207,21 +207,24 @@ def get_parameter_dtype(parameter: Union[nn.Module, GenerationMixin, "ModuleUtil
         # if no floating dtype was found return whatever the first dtype is
         return last_dtype
 
-    else:
-        # For nn.DataParallel compatibility in PyTorch > 1.5
-        def find_tensor_attributes(module: nn.Module) -> List[Tuple[str, Tensor]]:
-            tuples = [(k, v) for k, v in module.__dict__.items() if torch.is_tensor(v)]
-            return tuples
+    for t in parameter.buffers():
+        if torch.is_tensor(t):
+            return t.dtype
```

Comment:
Like for the others, it should only be returned if it's a floating point dtype as buffers may be bools or ints.